cql3/selection/selectable.cc

```diff
@@ -142,20 +142,19 @@ shared_ptr<selector::factory>
 selectable::with_field_selection::new_selector_factory(database& db, schema_ptr s, std::vector<const column_definition*>& defs) {
     auto&& factory = _selected->new_selector_factory(db, s, defs);
     auto&& type = factory->new_instance()->get_type();
-    auto&& ut = dynamic_pointer_cast<const user_type_impl>(type->underlying_type());
```

Comment:
Philosophical reason: checking if the result of `dynamic_cast` is null is a very implementation-focused way to check if something is a user type, which I don't like. But then there's a `static_cast` further down below instead, so not really much of an improvement...

I did it for efficiency more than anything else, since `dynamic_cast`s are slow. It probably doesn't matter here, but the fewer dynamic casts we have, the better I feel :P

Reply:
This will be converted to a visitor One Day (tm) anyway.

Reply:
Leaving it as it is